pkg/controller/networkpolicy/crd_utils.go

```diff
@@ -194,27 +196,39 @@ func (n *NetworkPolicyController) toAntreaPeerForCRD(peers []v1alpha1.NetworkPol
 			addressGroup := n.createAddressGroup(np.GetNamespace(), peer.PodSelector, peer.NamespaceSelector, peer.ExternalEntitySelector, nil)
 			addressGroups = append(addressGroups, addressGroup)
 		}
-		if peer.Scope == v1alpha1.ScopeClusterSet {
-			clusterSetScopeSelectors = append(clusterSetScopeSelectors, antreatypes.NewGroupSelector(np.GetNamespace(), peer.PodSelector, peer.NamespaceSelector, nil, nil))
+		if n.stretchNPEnabled && peer.Scope == v1alpha1.ScopeClusterSet {
+			newClusterSetScopeSelector := antreatypes.NewGroupSelector(np.GetNamespace(), peer.PodSelector, peer.NamespaceSelector, nil, nil)
+			*clusterSetScopeSelectors = append(*clusterSetScopeSelectors, newClusterSetScopeSelector)
+			labelIdentities = append(labelIdentities, n.labelIdentityInterface.AddSelector(newClusterSetScopeSelector, internalNetworkPolicyKeyFunc(np))...)
```

Comment:
It is not easy to understand the code that we add new selector in this func (which seems for generating NetworkPolicyPeer only) and remove stale selectors later. Is it possible to refactor the code to add/remove selectors in one place in the caller?

Reply:
I am afraid it is not straightforward to refactor the code so that the add/remove selectors are in one place. The reason is as follows: 

in a single policy, there can be multiple places where a standalone ClusterSet-scoped selector needs to be registered with the `label_group_index`: each peer in each ingress/egress rule can have a unique ClusterSet-scoped selector, not to mention that per-Namespace rules will cause a ClusterSet-scoped selector in a single peer to be split into multiple ones. Each such selector needs the `label_group_index` to tell them what are the label IDs matched specifically, so that the controller can translate those into correct Antrea controlplane rules. Hence, the addSelector needs to be called each time a ClusterSet-scoped selector is encountered.

On the other hand, it is only when the entire policy is processed that we can know what are the unique set of ClusterSet-scoped selectors that a policy has, and cross-check it with old cache and remove stale ones. This is necessary so that future Label Identity update events will only notify the correct set of policies that have ClusterSet-scoped selectors matching those labels.

If we were to add/remove selectors altogether, then the controller will need to take a first pass thru the policy, collect all possible ClusterSet-scoped selectors, register these with the `label_group_index` and remove stale ones, and only to loop thru the peers all over again later in order to get the label IDs of the selectors that it just registered. This will not be very efficient.

Back to the code readability front, do you think it will help if I refactor the above lines into:
```
newClusterSetScopeSelector := antreatypes.NewGroupSelector(np.GetNamespace(), peer.PodSelector, peer.NamespaceSelector, nil, nil)
*clusterSetScopeSelectors = append(*clusterSetScopeSelectors, newClusterSetScopeSelector)
matchedLabelIDs := n.labelIdentityInterface.AddSelector(newClusterSetScopeSelector, internalNetworkPolicyKeyFunc(np))
labelIdentities = append(labelIdentities, matchedLabelIDs...)
```

Reply:
Humm... I see the problem. Yes, it is at least a little better to make `AddSelector()` a separate line. Maybe add a comment to func `toAntreaPeerForCRD()` about it will add the selector to `n.labelIdentityInterface`.

Reply:
Refactored and added comment